runtime/themes/monokai_soda.toml

```
@@ -0,0 +1,122 @@
+# Monokai Soda port for Helix (https://helix-editor.com)
+# Author : Jimmy Zelinskie <jimmy@zelinskie.com>
+
+# Syntax
+
+## Constants
+"constant" = "white"
+"constant.builtin" = "pink"
+"constant.character.escape" = "blue"
+"constant.numeric" = "purple"
+
+## Diagnostics
+"diagnostic" = { modifiers = ["underlined"] }
+"diagnostic.error" = { underline = { style = "curl", color = "pink" } }
+"diagnostic.warning" = { underline = { style = "curl", color = "orange" } }
+"diagnostic.info" = { underline = { style = "curl", color = "white" } }
+
+## Diffs
+"diff.plus" = "green"
+"diff.delta" = "orange"
+"diff.minus" = "pink"
+
+## Functions
+"function" = "green"
+"function.macro" = "blue"
+"function.builtin" = "pink"
+"constructor" = "blue"
+
+## Git
+"git.delta.moved" = "orange"
+
+## Keywords
+"keyword" = "pink"
+"keyword.directive" = "blue"
+
+## Punctuation
+"punctuation" = "gray"
+
+## Strings
+"string" = "yellow"
+
+## Types
+"type" = "blue"
+"type.builtin" = "pink"
+
+## Variables
+"variable"  = "white"
+"variable.builtin" = "pink"
+"variable.other.member" = "white"
+"variable.parameter" = "softorange"
+
+## Markup
+"markup.heading" = "green"
+"markup.bold" = { fg = "orange", modifiers = ["bold"] }
+"markup.italic" = { fg = "orange", modifiers = ["italic"] }
+"markup.link.url" = { fg = "orange", modifiers = ["underlined"] }
+"markup.link.text" = "yellow"
+"markup.quote" = "green"
+
+## Misc
+"attribute" = "blue"
+"comment" = { fg = "gray", modifiers = ["italic"] }
+"error" = "pink"
+"hint" = "white"
+"info" = "white"
+"label" = "yellow"
+"module" = "softorange"
+"namespace" = "pink"
+"operator" = "pink"
+"special" = "softorange"
+"warning" = "orange"
+
+# Editor UI
+
+## Main
+"ui.background" = { bg = "background" }
+"ui.text" = "white"
+"ui.window" = { bg = "darkgray" }
+
+## Debug (TODO)
+
+## Menus
+"ui.menu" = { fg = "white", bg = "darkgray" }
+"ui.menu.selected" = { modifiers = ["reversed"] }
+"ui.popup" = { bg = "darkgray" }
+"ui.help" = { fg = "white", bg = "darkgray" }
+
+## Gutter
+"ui.linenr" = "darkgray"
+"ui.linenr.selected" = "orange"
+
+## Cursor
+"ui.cursor.primary" = { fg = "white", modifiers = ["reversed"] }
+"ui.cursor.match" = { fg = "white", modifiers = ["reversed"] }
+"ui.selection" = { bg = "darkgray" }
+
+## Statusline
+"ui.statusline" = { bg = "darkgray" }
+"ui.statusline.inactive" = { fg = "white", bg = "darkgray" }
+"ui.statusline.normal" = { fg = "white", bg = "blue" }
+"ui.statusline.insert" = { fg = "white", bg = "green" }
+"ui.statusline.select" = { fg = "white", bg = "purple" }
+
+"ui.text.focus" = { fg = "yellow", modifiers = ["bold"] }
+"ui.virtual.whitespace" = "darkgray"
```

```suggestion
"ui.virtual" = "darkgray"
```

Other than rulers, the other `ui.virtual` scopes all benefit from faint foreground colors like this. I'm suggesting this just so that others with inlay-hints for example get sane colors for those (though you might not use inlay hints yourself)